impress-org/givewp
# feature: Add "ProTip" message to Reports page

## Description

  This PR introduces an enhancement aimed at providing users with additional information about our Recurring and Fee Recovery addons, especially those who may not be aware of it. The primary feature is a "ProTip" message that will be displayed on the Reports page. The message includes a UTM link that, when clicked, will open the add-on's documentation page in a new window.

To ensure a seamless user experience, the PR includes functionality to track and save the user's interaction with the options. This information will be stored in the database under the options table. If a user has already installed a specific add-on, it will be automatically removed from the list of options to display, preventing redundant or unnecessary messages from appearing.

### Options list includes: Addon - option _name
- Recurring Donations addon -  'givewp_reports_recurring_recommendation_dismissed
- Fee Recovery addon -  givewp_reports_fee_recovery_recommendation_dismissed',


## Affects

Reports Page

## Visuals
https://www.loom.com/share/7e5cc6f67e6a4af5bd0e34be318a6b50


## Testing Instructions
- Install and activate Recurring addon
- Activate the addon and visit the Page to confirm that the "ProTip" message does not appear.
- Deactivate the addon and view the Reports Page to confirm that the message does appear.
- Close the message via the exit icon on the right, visit the Reports page and refresh the page to confirm the message no long appears.


## Pre-review Checklist
